src/constants/autocomplete.ts

```diff
@@ -0,0 +1,7 @@
+export const AUTOCOMPLETE_CHANGE_REASON = {
```

Comment:
I have questions about the need for these here.

First it's the quesion: do we really need an abstraction for this when we only use half of the fields once? Is there some Type.js advantage over using a type here? I don't see any "default" branch for the switch, so it doesn't enforce all possible branches of the struct are implemented -- so why the struct?


Second issue is about this `src/constants` folder - I asked for this `constants` folder is here to allow non-JS devs to change small aspects such as label wording, field lists, icons, etc. This is not documented in the source, so maybe we should add a small note in the `constants` folder with this detail?

In any case I wouldn't bloat this folder too much with internal React details; you could make these consts local to that file.

Reply:
I initially thought about creating a local `constant.ts` file for that specific folder. Then, I thought about future reusability of these values, in case there would be any other autocomplete implementation that would require these fields. This abstraction does not bring any benefit other then readability and reusability on the long term. 

1. I could remove this entirely and use the string values directly in the code.
2. Move this from `src/constants` to its `./constants.ts` local to that implementation. (as you mentioned in your last statement)

Since this is subjective, please state your preferance and I would make the change accordingly :)

Reply:
I wouldn't make these constants when the strings are only ever used once - when it starts repeating itself a lot, then it makes sense to DRY it

I wouldn't expect them to change this specific thing again in the future either - this is probably one of those 200+ deprecation warnings we have on any build of the UI - just worse, because we don't get deprecation warnings for internally changing strings. If anything they'll change everything else but this.

So just throw in the strings now - and if you repeat the strings a lot in the future, you'll possibly find a simpler way of abstracting them out, i.e. functions, classes, subcomponents, etc.

Reply:
I would also put a link as a comment inside the swich, pointing to docs that lists all those possible values for the string - I feel that's of more value into understanding than adding opaque const names to the opaque strings